icu4c/source/test/intltest/ustrtest.cpp

```diff
@@ -2408,48 +2412,176 @@ void UnicodeStringTest::TestLargeAppend() {
     }
 }
 
-// TODO: copy as 'w' version
 void UnicodeStringTest::TestU16StringView() {
     IcuTestErrorCode status(*this, "TestU16StringView");
-    // TODO: real testing...
+    // ICU-22843 Test ICU 76 new UnicodeString APIs that take or return a std::u16string_view
+    // or something convertible to it.
+    // NOTE: Keep this function very parallel with TestWStringView()!
     const char16_t *p16 = u"p16";
     std::u16string_view sv16 = u"sv16";
     std::u16string str16 = u"str16";
 
-    UnicodeString any(true, u"any", 3);
-    UnicodeString fromPtr(p16);
-    UnicodeString fromSV(sv16);
-    UnicodeString fromSV2(u"sv16_2"sv);  // u16string_view literal
-    UnicodeString fromStr(str16);
+    // These copy the string contents.
+    UnicodeString fromPtr(p16);  // pointer is convertible to std::u16string_view
+    UnicodeString fromSV(sv16);  // std::u16string_view itself
+    UnicodeString fromSV2(u"sv16_2"sv);  // std::u16string_view literal
+    UnicodeString fromStr(str16);  // std::u16string is convertible to std::u16string_view
+    assertEquals("UnicodeString(const char16_t *pointer)", UnicodeString(u"p16", 3), fromPtr);
```

Comment:
Would it make sense to do `UnicodeString(u"p16", -1)` instead of `UnicodeString(u"p16", 3)` here and in all the analogous places?  I'm a little worried somebody will change the string contents and forget they also have to change the length.

Reply:
> If you don't mind, I would like to squash and then merge as is.

That's fine with me.